pkg/controller/account/account_controller.go

```diff
@@ -285,57 +285,57 @@ func (r *ReconcileAccount) Reconcile(request reconcile.Request) (reconcile.Resul
 	// If the account is BYOC, needs some different set up
 	if currentAcctInstance.Spec.BYOC {
 		reqLogger.Info("BYOC account")
-
-		// Create client for BYOC account
-		byocAWSClient, accountClaim, err := r.getBYOCClient(currentAcctInstance)
-		if err != nil {
-			if accountClaim != nil {
-				r.accountClaimBYOCError(reqLogger, accountClaim, err)
-			}
-			return reconcile.Result{}, err
-		}
-
-		err = validateBYOCClaim(accountClaim)
-		if err != nil {
-			r.accountClaimBYOCError(reqLogger, accountClaim, err)
-			return reconcile.Result{}, err
-		}
-
-		// Ensure the account is marked as claimed
-		if currentAcctInstance.Status.Claimed != true {
-			reqLogger.Info("Marking BYOC account claimed")
-			currentAcctInstance.Status.Claimed = true
-			return reconcile.Result{}, r.statusUpdate(reqLogger, currentAcctInstance)
-		}
-
-		// Create access key and role for BYOC account
-		if currentAcctInstance.Status.State == "" {
-			// Rotate access keys
-			err = r.byocRotateAccessKeys(reqLogger, byocAWSClient, accountClaim)
+		if currentAcctInstance.Status.State == "" || currentAcctInstance.Status.Claimed != true {
+			// Create client for BYOC account
+			byocAWSClient, accountClaim, err := r.getBYOCClient(currentAcctInstance)
 			if err != nil {
-				reqLogger.Info("Failed to rotate BYOC access keys")
-				r.accountClaimBYOCError(reqLogger, accountClaim, err)
+				if accountClaim != nil {
+					r.accountClaimBYOCError(reqLogger, accountClaim, err)
```

Comment:
I don't think we want to error here right? If a reconcile for the AccountClaim gets triggered we may just want to ignore it. What is the reason for the error is it an issue?

Reply:
This already exists in master today, line 292 of the account_controller. The only change in this PR is that its within a new if statement above.